### ISSUE TITLE
GitHub Provider Not Working?

### ISSUE THREAD

**Issue by RickStrahl** [TerribleDev/OwinOAuthProviders#143]
I'm getting an error from the GitHub provider that seems to indicate that the Redirect URL is incorrect. I also have Twitter and Google hooked up and both of those seem to be working, but the 

I've double checked my end and I'm sure the URLs and auth tokens are valid in my app. Running MVC 5 application with the latest NuGet package of the providers installed (1.26) and I have the following in startup:

``` csharp
if (!string.IsNullOrEmpty(App.Secrets.GitHubClientId))
{
    app.UseGitHubAuthentication(App.Secrets.GitHubClientId, App.Secrets.GitHubClientSecret);
}
```

I've verified the keys are correct on startup.

In the GitHub developer API I've ensured that the URL points to:

```
http://codepaste.net/signin-github
```

and I can see that URL fired when tracing with Fiddler.

The URL response I get is:

```
http://codepaste.net/signin-github?error=redirect_uri_mismatch&error_description=The+redirect_uri+MUST+match+
the+registered+callback+URL+for+this+application.
&error_uri=https%3A%2F%2Fdeveloper.github.com%2Fv3%2Foauth%2F%23redirect-uri-mismatch
&state=...
```

Here's what the Fiddler Trace looks like:

![image](https://cloud.githubusercontent.com/assets/1374013/12833632/f1539fe6-cb45-11e5-8466-e5788b3ce736.png)

It looks like it all goes Ok until the last redirect.

This also worked when I initially set this up last year, but somewhere in that timeframe the same code started no longer working.

Any ideas what I'm missing and how I can track this down? Comparing to the Google Login it all seems to go fine as far as the Controller code is concerned. It's just that there appears to be some issue with the last redirect.

**Comment by jerriep** [comment]
Hey Rick,

I am a bit confused as the screenshot of the Fiddler trace is different though... It shows that GH is indeed returning a code to you which the middleware should then be exchanging for an access token.

Am I missing something?

**Comment by RickStrahl** [comment]
Hmmm... Ok, so I looked closer at my code and it turns out you are absolutely correct. The final error was actually being caused by my internal code which was rejecting the token because I was trying to log in with a non-existing token (or another external provider token already attached) rather than creating a new token. Once I cleared out all the saved tokens and started from scratch it all worked as expected.

So it is working if I completely unlink my app account, then relink the app account to GitHub. 
Sorry for the false alarm...

I ended up here because a number of people mentioned they couldn't log in with GitHub or couldn't associate their account with GitHub. It's possible the update to the latest rev fixed that and then my not unlinking (from Google first) caused the problem.
